### ISSUE TITLE
SCM: too many "Publish Changes" buttons in multi root workspace

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#133950]
Opening a multi root workspace from folders that are not published, I find the look is pretty overwhelming now:

![image](https://user-images.githubusercontent.com/900690/135047262-1bc36952-dd2a-4f19-9ea2-432fee4954f7.png)

//cc @lszomoru

**Comment by jrieken** [comment]
Even the single repo case is soo much more noisy now. There is the bold "vscode" title with a bunch of command and opposed to that a ton of whitespace above that

| after | before |
|---|---|
|<img width="422" alt="Screen Shot 2021-09-28 at 10 12 38" src="https://user-images.githubusercontent.com/1794099/135049346-89e8ed61-4f3a-43c2-a546-9cfae530d0c3.png">|<img width="334" alt="Screen Shot 2021-09-28 at 10 12 19" src="https://user-images.githubusercontent.com/1794099/135049243-9900b09f-51ac-480c-a168-ccef88be1da5.png">|

**Comment by gjsjohnmurray** [comment]
@jrieken I think that's caused by the change to the default for the `scm.alwaysShowRepositories` setting.

**Comment by jrieken** [comment]
Thanks @gjsjohnmurray !

**Comment by gjsjohnmurray** [comment]
@bpasero did you consider opting out of the new button with `"git.showUnpublishedCommitsButton": "never"` ?

**Comment by bpasero** [comment]
I know there is a setting but this is the default experience I am worried about for all users.

**Comment by jrieken** [comment]
> default experience I am worried about for all users.

Same goes for the default of `scm.alwaysShowRepositories` - we pride ourself to have minimal UI, little borders/chrome etc and this seems like an unneeded complication of things

**Comment by gjsjohnmurray** [comment]
> Same goes for the default of `scm.alwaysShowRepositories`

Fair point. OTOH I recall at least one issue raised by someone who hadn't spotted the status bar panel and couldn't work out how to switch branch.

Re the new action button, maybe hide it when multiple repos are being displayed. Or give the user an easy way to turn it off on the spot.

**Comment by jrieken** [comment]
There also seems to be glitch with unicode characters

<img width="433" alt="Screen Shot 2021-09-28 at 14 02 04" src="https://user-images.githubusercontent.com/1794099/135083035-925376eb-8e8c-4433-ba6d-ff5bdcc452f7.png">

**Comment by misolori** [comment]
Just so it's clear, this issue is about the "publish changes" button and #134015 is about the default experience for the SCM view

**Comment by gjsjohnmurray** [comment]
Instead of the big blue action buttons, could the git extension use the prompt text of the input box to guide the user? The guidance is mostly useful at a point where there are no staged changes, so the user doesn't have any reason to type text into the input box.

For example, instead of:

![image](https://user-images.githubusercontent.com/6726799/135245643-23f9c4f9-cffb-43bf-8245-3004afe4d8c3.png)

we could have:

![image](https://user-images.githubusercontent.com/6726799/135247693-243971b0-e42d-4347-9f84-16d213ed98a3.png)

(I mocked this up using developer tools and didn't find a way to put the actual cloud-upload codicon into the prompt)

**Comment by gjsjohnmurray** [comment]
> Instead of the big blue action buttons, could the git extension use the prompt text of the input box to guide the user?

For bonus points, make the prompt text sensitive to whether or not there **is** an upload/sync button above. That way we could revert the controversial recent change that has made `scm.alwaysShowRepositories` true by default, and instead rely on the prompt text to draw a new user's attention to the status bar panel contributed by the built-in git extension.

**Comment by JacksonKearl** [comment]
Another option I hacked out of CSS just now: 
![image](https://user-images.githubusercontent.com/8586769/135517125-4a864dd8-27dc-4c6d-aee8-fa51cb938520.png)

Basically just taking the existing status bar text and making it close to the SCM input (it always feels weird to do my commit stuff in one area of the screen then need to go to a totally different area to sync/publish/whatever)

**Comment by bpasero** [comment]
@lszomoru This is fixed right?

**Comment by lszomoru** [comment]
Yes. We now only select one repository to be shown in the Changes view.
